### ISSUE TITLE
Ambiguity with name lookup into namespace with inline namespaces not detected

### ISSUE THREAD

**Issue by be579e8c-b74b-475b-a126-3b8cefdd833c** [llvm/llvm-project#10733]
|  |  |
| --- | --- |
| Bugzilla Link | [10361](https://llvm.org/bz10361) |
| Version | trunk |
| OS | Linux |
| CC | @DougGregor,@gnzlbg,@griwes,@nlewycky |

## Extended Description 
This ambiguity is not detected by clang. Apparently it stops looking into 'A' when it finds the direct declaration of 'f' in A, and doesn't look into the inline namespace. The FDIS says that name lookup for A::f yields both A::f and A::B::f. 
```c++
namespace A {
inline namespace B {
  template<typename A> struct f { };
}
template<typename A>
struct f { };
}
 
int main() {
  A::f<int>();
}
```

**Comment by philnik777** [comment]
Fixed in Clang 3.5: https://godbolt.org/z/oPfn6v1aK
